src/sentry/profiles/task.py

```diff
@@ -186,6 +215,11 @@ def _symbolicate_profile(profile: Profile, project: Project) -> bool:
                 reason="profiling_failed_symbolication",
             )
             return False
+        profile["debug_meta"]["images"] = []
+        for imgs in images.values():
+            profile["debug_meta"]["images"] += imgs
```

Comment:
this is a bit weird, it reorders the `images`, and discards anything thats not `sourcemap` or `macho`.
better to just save this in a temporary variable if you indeed need to overwrite the value in `profile` temporarily?

Reply:
Why do we overwrite it at all? Is there a reason we can't just pass on the `images` list as is?

Reply:
symbolicator expects to receive a pre-filtered list of debug images, thats why you have two filtered down subsets.

Reply:
@Swatinem as discussed offline this would work as it's only used withing `_symbolicate_profile` and therefore for `JS` or `Cocoa`.

Anyway I've modified the `get_debug_images_for_platform` function to use `native_images_from_data` function that normal symbolication uses to filter the objects as you suggested so that it's not too restrictive.

Reply:
I still don't understand why we have to modify the images list on the profile. This is happening after symbolication has been completed. You've already extracted the subset of images you need for each request to symbolicator so there's no reason to modify the images list on the profile.

Reply:
@Zylphrex the debug images are modified by `Symbolicator`. It adds some status, etc.

We used do update the profile images even before this PR. [Here](https://github.com/getsentry/sentry/blob/master/src/sentry/profiles/task.py#L416-L417) in the `_process_symbolicator_results` function.


Now, do we use `debug_images` at all to provide profiling info to the user? not really, but I'd like to keep it for 2 reason:

1. it might still be useful for debugging when we download the raw profile (which contains this returned debug images)
2. up until today we kept the `debug_images` returned by Symbolicator instead of the original ones and I'm not too keen on introducing this change here since it's unrelated to the purpose of this PR

Reply:
I see, I guess then I have 1 more question, can there be images other than sourcemaps and native images in this list that we're discarding here but is useful? Should we try to keep them?

And as a nit, `+=` will make a copy of the list each iteration, you should use `.extend(...)`.

Reply:
Considering we're inside the `_symbolicate_profile` function yes, I think that `sourcemaps` and `native` images are all the images that there should be.

@Swatinem could you confirm please?

Reply:
That was my main point, I believe there can be more than just those two sets of images, as your filters are very restrictive.
At least the native processor does have quite some logic around merging the original image list with whatever symbolicator is returning. However it also does things like create these error banners about missing debug images, etc. That is probably something you don’t want to have for profiling.

Long story short, I still think its best to filter the list and then update the filtered images after symbolication.

this is how its done in the native symbolication pipeline:

```
    modules = native_images_from_data(data)

…

    assert len(modules) == len(response["modules"]), (modules, response)

    os = get_os_from_event(data)

    for raw_image, complete_image in zip(modules, response["modules"]):
        _merge_image(raw_image, complete_image, os, data)
```

I don’t think you want to just copy all of that, as the `os` handling also is primarily related to conditionally displaying that "debug file not found" warning.

Reply:
@Swatinem I've just pushed a commit that preserve the original list of images and updates the filtered images through `_merge_image` as you suggested.
This should address the concern that both you and @Zylphrex brought up.